tests/integration/test_social_media_services.py

```diff
@@ -0,0 +1,112 @@
+# Standard Stuff
+import json
+
+# Third Party Stuff
+import pytest
+import tweepy
+from unittest import mock
+
+# nexus Stuff
+from tests import utils as u
+from django.urls import reverse
+from django.utils import timezone
+from tests import factories as f
+from nexus.social_media import services
+from nexus.social_media.models import Post
+from settings.development import TWITTER_OAUTH
+from nexus.base.exceptions import WrongArguments
+from django.core.files.uploadedfile import SimpleUploadedFile
+
+
+pytestmark = pytest.mark.django_db
+
+
+def test_update_post_object(client):
+    user = f.create_user(email='test@example.com', password='test')
+    client.login(user=user)
+
+    url = reverse('posts-list')
+    post = {
+        'text': 'Announcement!',
+        'posted_at': 'twitter',
+        'scheduled_time': '2018-10-09T18:30:00Z'
+    }
+
+    response = client.json.post(url, json.dumps(post))
```

Comment:
first check if your call was successful here

`assert response.status_code == 200`

and then check if you get the appropriate error or appropriate keys in the data.

```
expected_keys = {'text', 'posted_at', 'scheduled_time'}
assert expected_keys.issubset(response.data.keys())
```

Reply:
But, I am not testing the API here. Just wanted to create a `Post` so that I can send that to a service `update_post_object` to see if the service actually does the job of updating a `Post` instance or not.
Shall I use factories to create a post, if I don't want to check the API?

Reply:
For services being used by APIs, I'd suggest testing the APIs.

That is why we're calling it integration tests.

Since this service is pretty small, it doesn't make sense to test it individually. However, for services spinning more than a few tens of lines, you can test them individually.

Reply:
I'm not sure, but can you check if `client.post(url, post)` works?